qtfred/src/mission/dialogs/CampaignEditorDialogModel.h

```diff
@@ -0,0 +1,257 @@
+#pragma once
+
+#include "mission/dialogs/AbstractDialogModel.h"
+#include "ui/dialogs/CampaignEditorDialog.h"
+#include "AssociatedPlainTextDocument.h"
+#include "CheckedDataListModel.h"
+#include <mission/missioncampaign.h>
+#include <ui/widgets/sexp_tree.h>
+
+#include <QTextDocument>
+#include <QPlainTextEdit>
+
+namespace fso {
+namespace fred {
+namespace dialogs {
+
+static const QString qstrEmpty{};
```

Comment:
Qt doesn't provide an empty string constant to which I could return references where necessary in the following, so I made one here.
I did not make it a member of `CampaignEditorDialogModel`, because conceptually, it doesn't have anything to do with that class. I can remove the static keyword, it doesn't make much sense here.